R/load_runner.R

```diff
@@ -23,20 +23,25 @@
 #' @keywords internal
 #'
 run_load <- function(con, load, extracted_data, transformed_data, test_queries,
-                     pre_load, post_load, path, test_file, dry_run, log_table,
-                     comment) {
+                     pre_load, post_load, path, test_file, transaction, dry_run,
+                     log_table, comment) {
   if (is.null(transformed_data)) {
     stop("Cannot run tests as no data has been transformed.")
   }
   log_data <- build_log_data(path, comment)
   verify_log_table(con, log_table, log_data)
   verify_first_run(con, log_table, log_data)
-  DBI::dbBegin(con)
+  if (transaction || dry_run) {
```

Comment:
perhaps do

```
use_transaction <- transaction || dry_run
```
above this and then use that value in both checks

Reply:
thanks